### PR TITLE
Fix several issues with update_tmserver

### PR DESCRIPTION
**Pull request by unho**

### CODE REVIEW

File: pootle/apps/pootle_app/management/commands/update_tmserver.py [hunk 1]
```diff
@@ -200,8 +201,8 @@ def _initialize(self, *args, **options):
         try:
             self.tm_settings = settings.POOTLE_TM_SERVER[options.get('tm')]
         except KeyError:
-            raise CommandError('Specified Translation Memory is not defined '
-                               'in POOTLE_TM_SERVER.')
+            raise CommandError("Translation Memory '%s' is not defined in "
+                               "POOTLE_TM_SERVER setting." % options.get('tm'))
```
Comment (dwaynebailey) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: This is a ``ImproperlyConfigured`` not a ``CommandError``

File: pootle/apps/pootle_app/management/commands/update_tmserver.py [hunk 2]
```diff
@@ -160,12 +161,12 @@ class Command(BaseCommand):
...
                          "be guessed for any of the input files."),
-        make_option('--project',
+        make_option('--display-name',
                     action='store',
                     dest='project',
                     default='',
-                    help='Project to use when displaying TM matches for this '
-                         'translations.'),
+                    help='Name to use as provenance when displaying TM '
+                         'matches for these translations.'),
```
Comment (dwaynebailey) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: You like the word provenance :)

Name used when displaying TM matches for these translations.

File: docs/server/commands.rst [hunk 3]
```diff
@@ -431,13 +431,18 @@ no actual data will be loaded or deleted (the TM will be left unchanged):
 
 This command also allows to read translations from files and build the TM
 resources in the external TM server. In order to do so it is mandatory to
-provide the :option:`--tm` and :option:`--project` options, along with some
-filename argument to import translations from. The provided project name will
-be displayed on TM matches in the translation editor:
+provide the :option:`--tm` and :option:`--display-name` options, along with
+some filename argument to import translations from.
```
Comment (dwaynebailey) [inline docs/server/commands.rst]: ...along with some files to import.

File: docs/server/commands.rst [hunk 4]
```diff
@@ -431,13 +431,18 @@ no actual data will be loaded or deleted (the TM will be left unchanged):
...
+
+The display name is a label used to group translations within a TM. A given TM
+can host translations for several display names. A use for the display name
+might be to specify the project from which the translation comes from. The
+display name will be shown on TM matches in the translation editor. Just
+specify it with :option:`--display-name`:
```
Comment (dwaynebailey) [inline docs/server/commands.rst]: The display name can be used to specify the name of the project from which the translations originate.

To specify a name use --display-name

File: pootle/apps/pootle_app/management/commands/update_tmserver.py [hunk 5]
```diff
@@ -129,6 +129,7 @@ def get_unit_data(self, unit):
 
 class Command(BaseCommand):
     help = "Load Translation Memory with translations"
+    args = "[files]"
```
Comment (dwaynebailey) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: Don't we also allow no files?  Is it possible to have more than one usage?

**Comment by dwaynebailey** [comment]
Some issues otherwise LGTM

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: IMHO it is better to display the user a specific message telling what is wrong than throwing a traceback that means nothing to the user.

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 5 above)
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: `files` is enclosed in `[]` to tell that this is optional.

Not sure what you mean with more than one usage.

Comment (dwaynebailey) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: The non-file version.  I just wanted to flag this rather then discuss it.

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 2 above)
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: Changing it.

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: So I am keeping this as it is for the reason in my previous comment.

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 5 above)
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: I think that `[files]` clearly states that files are optional, so this stands for both file and non-file versions.

Displaying two different usage snippets (both at the same time) would require to override the `usage()` method, but I don't think it is actually necessary.

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 1 above)
Comment (dwaynebailey) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: Cool I'm fine with that logic.

**Comment by dwaynebailey** [comment]
LGTM

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 1 above)
Comment (julen) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: So what about inviting users to review their settings as well? Something like *"Translation Memory '%s' is not defined in the POOTLE_TM_SERVER setting. Please ensure it exists and double-check you typed it correctly."*

**Comment by julen** [comment]
Thank you for taking this @unho, GTM.

File: pootle/apps/pootle_app/management/commands/update_tmserver.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_app/management/commands/update_tmserver.py]: Yes, it is better. Changed.

**Pull request merged**